pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,222 @@
+Docker image. Also check your agents have `kubectl` access to your target
+cluster. Refer to the notes at the end of tutorial for tips on setting this
+up.
+
+<%= image "final_test_pipeline.png", width: 736, height: 199, alt: 'Final Test Pipeline' %>
+
+Let's go.
+
+## Create the Deploy Pipeline
+
+This section covers creating a new Buildkite pipeline that loads steps from
+`.buildkite/pipeline.deploy.yml`. Deploy pipelines are like any other pipeline
+except they're triggred by events other than code pushes. We'll use a [trigger
+steps](https://buildkite.com/docs/pipelines/trigger-step) later on to connect
+the test and deploy pipelines.
+
+Create a [new pipeline](https://buildkite.com/new) for your repo. Use
+`buildkite-agent pipeline upload .buildkite/pipeline.deploy.yml` for the first
+step as shown in the image.
+
+<%= image "new_pipeline.png", width: 939/2, height: 454/2, alt: 'Creating a New Pipelien'  %>
+
+Now create `.buildkite/pipeline.deploy.yml` with a single step. We'll write the
+deploy script in the next step.
+
+```yml
+steps:
+  - label: "\:rocket\: Push to \:kubernetes\:"
+    comamnd: script/buildlkite/deploy
+    concurrency: 1
+    concurrency_group: deploy/tutorial
+```
+
+Set `concurrency` and `concurrency_group` when updating mutable state. These
+settings ensures only one step runs at a time, effectively creating a deploy
+queue.
+
+## Writing the Deploy Script
+
+The next step is writing a deploy script that generates a [Kubernetes deployment
+manifest][k8s_deployment] from the `DOCKER_IMAGE` environment variable.
+
+Let's start with manifest file. This sample file creates a Deployment with
+three replicas (horizontal scale in Kubernetes lingo) each listening port
+`3000`. Change the `containerPort` to fit your application.
+
+<section class="Docs__note">
+  <p>
+    The <a
+    href="https://kubernetes.io/docs/concepts/workloads/controllers/deployment/">official
+    deployment documentation</a> covers much more than what fits in this
+    tutorial. Refer back to these docs for information on setting CPU and
+    memory, controlling networking, deployment update strategies, and how to
+    expose your application to the internet.
+  </p>
+</section>
+
+Let's call this file `k8s/deployment.yml`.
+
+```yml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tutorial
+  labels:
+    app: tutorial
+spec:
+  # TODO: replace with a value that fits your application
+  replicas: 3
+  selector:
+    matchLabels:
+      app: tutorial
+  template:
+    metadata:
+      labels:
+        app: tutorial
+    spec:
+      containers:
+        - name: app
+          image: "${DOCKER_IMAGE}"
+          ports:
+            # TODO: replace with the correct port for your application
+            - containerPort: 3000
+```
+
+Note manifest includes `${DOCKER_IMAGE}`. There is no environment variable
+subsitution in YAML or `kubectl` itself. This is where the deploy script comes
```

Comment:
If we could somehow specify that the deploy script we're talking about here is related to the Buildkite pipeline, that would be super. No specific suggestions though 😅